sources/packages/backend/libs/integrations/src/cas/cas.service.ts

```diff
@@ -267,6 +246,28 @@ export class CASService {
     return supplierAddress;
   }
 
+  /**
+   * Handles bad request errors from CAS API.
+   * @param error The error object caught during the request.
+   * @param defaultMessage The default message to throw for non-specific errors.
+   */
+  private handleBadRequestError(error: unknown, defaultMessage: string): never {
+    if (
+      error instanceof AxiosError &&
+      error.response?.status === HttpStatus.BAD_REQUEST &&
+      !!error.response?.data[CAS_RETURNED_MESSAGES]
+    ) {
+      const casKnownErrors =
+        error.response.data[CAS_RETURNED_MESSAGES].toString();
+      throw new CustomNamedError(
+        "CAS Bad Request Errors",
+        CAS_BAD_REQUEST,
+        casKnownErrors.split("|"),
```

Comment:
Was the API response checked to ensure that there are no white spaces between the "|"?

Reply:
Nice one, i missed it, it was with spaces

`{
    "CAS-Returned-Messages": "[0034] SIN is already in use. | [9999] Duplicate Supplier , Reason: [0065]- Possible duplicate exists, please use online form"
}`

Reply:
As mentioned below, let's try to have a unit test to validate the error in some of the next PRs.
https://github.com/bcgov/SIMS/pull/4017#discussion_r1872276210